cql3/restrictions/statement_restrictions.cc

```diff
@@ -1394,6 +1395,94 @@ bool is_satisfied_by(
+            return false;
+        } else {
+            // *this is a tighter upper bound if it's smaller.
+            return _value_type.compare(*this->_value, *that._value) < 0;
+        }
+    }
+
+    bytes_view_opt value() const {
+        if (_unbounded) {
+            throw std::logic_error("bound_t::value() called on unbounded");
+        }
+        return _value;
+    }
+};
+
+bound_t get_bound(const expression& restr, const query_options& options, statements::bound bnd) {
+    return std::visit(overloaded_functor{
+            [&] (const conjunction& conj) {
+                if (conj.children.empty()) {
+                    throw std::logic_error("Empty conjunction");
+                }
+                auto invoke_get_bound = [&] (const ::shared_ptr<expression>& p) {
+                    return get_bound(*p, options, bnd);
+                };
+                // All conjunction elements have the same LHS; this is how
+                // single_column_restrictions::add_restriction constructs it.
+                std::vector<bound_t> children_bounds(
+                        boost::make_transform_iterator(conj.children.cbegin(), invoke_get_bound),
+                        boost::make_transform_iterator(conj.children.cend(), invoke_get_bound));
+                if (is_start(bnd)) {
+                    return *boost::max_element(children_bounds, [&] (const bound_t& a, const bound_t& b) {
+                        return b.is_tighter_lb_than(a);
+                    });
+                } else {
+                    return *boost::min_element(children_bounds, [&] (const bound_t& a, const bound_t& b) {
+                        return a.is_tighter_ub_than(b);
+                    });
+                }
+            },
+            [&] (const binary_operator& opr) {
+                // lhs must be column_values; CQL forbids token.
+                auto cv = std::get<0>(opr.lhs);
+                if (cv.size() != 1) {
+                    throw std::logic_error("get_bound invoked on multi-column restriction");
+                }
+                const auto cmptype = comparator(cv[0]);
+                static const std::vector<std::vector<const operator_type*>> operators{
+                    {&operator_type::EQ, &operator_type::GT, &operator_type::GTE}, // These mean a lower bound.
+                    {&operator_type::EQ, &operator_type::LT, &operator_type::LTE}, // These mean an upper bound.
+                };
+                const auto zero_if_lower_one_if_upper = get_idx(bnd);
+                if (boost::algorithm::any_of_equal(operators[zero_if_lower_one_if_upper], &opr.op)) {
+                    return bound_t(*cmptype, to_bytes_opt(opr.rhs->bind_and_get(options)));
+                }
+                return bound_t(*cmptype);
+            },
+            [] (auto& default_case) {
+                throw exceptions::unsupported_operation_exception("Unknown wip::expression subtype");
+            }
```

Comment:
Isn't it better to leave default_case out and let the compiler complain that we forgot to implement it?

Reply:
Done.